app/api/sync/routes.ts

```diff
@@ -71,7 +71,22 @@ export default (app: Application) => {
   app.post(
     '/api/sync/upload',
     needsAuthorization(['admin']),
-    uploadMiddleware(uploadsPath, storage),
+    async (req, res, next) => {
+      await new Promise((resolve, reject) => {
+        multer({ storage }).single('file')(req, res, err => {
+          if (!err) resolve('ok');
+          reject(err);
+        });
+      });
```

Comment:
This is required to inject the proper values into the REQ object, otherwise, the values are not directly accessible. But this is rendered useless if we go with the multi-route approach. Thanks.